### ISSUE TITLE
Add watchdog to the list of dependencies

### ISSUE THREAD

**Issue by severo** [huggingface/doc-builder#251]
Currently, after installing doc-builder (I did it with version 0.3.0 from pip, but I understand it has not changed), if I run `doc-builder preview`, I get:

```
ImportError: Please install `watchdog` to run `doc-builder preview` command.
You can do so through pip: `pip install watchdog`
```

`watchdog` should be added to setup.py

**Comment by sgugger** [comment]
No, it's not a hard-dependency because it's only needed for `doc-builder preview`. We could add an extras for preview however with this dependency.
